silhouette/app/com/mohiva/play/silhouette/api/services/AuthenticatorService.scala

```diff
@@ -128,7 +150,7 @@ trait AuthenticatorService[T <: Authenticator] {
    * @param authenticator The authenticator to touch.
    * @return The touched authenticator on the left or the untouched authenticator on the right.
    */
-  protected[silhouette] def touch(authenticator: T): Either[T, T]
+  def touch(authenticator: T): Either[T, T]
```

Comment:
You must look through the complete code of the Silhouette class. The `handleAuthentication` method returns also an `Either[Authenticator, Authenticator]`for the old or maybe a new authenticator. So the `extract` method is used five times in this class. Sure we could also use `fold(identity, identity)` directly, but I thought the extract method is a bit shorter.

Remember also that the `touch` method pertains to the public API of the authenticator service. Just because you don't see my statement 'more fluent interface' valid in the scope of the Silhouette class doesn't mean it isn't valid outside of this class.

I don't really see your problem here. If it returns `Option[A]` or `Either[A, A]`, for me it is nearly the same. I see benefits by returning either the unmodified or the modified, so I can work always with the result of the function and not in case of the unmodified with another variable. For me it's more the functional way.